modules/javafx.graphics/src/main/java/com/sun/javafx/scene/TreeShowingExpression.java

```diff
@@ -41,14 +41,13 @@
  * This class provides the exact same functionality as {@link NodeHelper#isTreeShowing(Node)} in
  * an observable form.
  */
-public class TreeShowingExpression extends BooleanExpression {
+public class TreeShowingExpression extends ReadOnlyBooleanPropertyBase {
```

Comment:
Maybe you could rename the class to `TreeShowingProperty`?

Reply:
Yeah, that would definitely be better.

Reply:
In `Node` we have `TreeVisiblePropertyReadOnly`. Best to align it with the other properties and remove the "ReadOnly" part.